app/content/webviewPreload.js

```diff
@@ -2,10 +2,29 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-var ipc = require('electron').ipcRenderer
 var messages = require('../../js/constants/messages')
 var KeyCodes = require('../../js/constants/keyCodes')
 
+var ipc = {
+  events: {},
+  embedder: null,
+  processMessage: function (event) {
+    if (event.origin === 'file://') {
```

Comment:
Can these origin checks be for `file:///full/path/to/index.html` instead of just `file://`? Otherwise any HTML file that the user has downloaded and opened in a tab/iframe could set itself as the embedder.